components/date-picker/private/calendar.jsx

```diff
@@ -220,7 +220,8 @@ const DatepickerCalendar = createReactClass({
 
 			// create new weeks
 			firstDayOfWeek = DateUtil.addWeeks(firstDayOfWeek, 1);
-			done = count++ > 2 && monthIndex !== firstDayOfWeek.getMonth();
+			done = count > 2 && monthIndex !== firstDayOfWeek.getMonth();
+			count += 1;
```

Comment:
This is a great example of why to try to avoid the postfix operator. In the old code here, `>` has a higher operator precedence than `++` but the code is written in a way that it _looks like_ `count` is incremented _before_ checking if its greater than `2`. However `>` get compared _first_, so `count` gets "greater than 2" _first_, and _then_ gets incremented _after_ the comparison which makes the old version of this code potentially confusing.